framework/src/headers/ip/v6/mod.rs

```diff
@@ -188,25 +188,36 @@ impl Ipv6Header {
     #[inline]
     pub fn flow(&self) -> Option<FlowV6> {
         if let Some(next_hdr) = self.next_header() {
-            let src_ip = self.src();
-            let dst_ip = self.dst();
-            if self.payload_size(0) >= 4 {
-                unsafe {
-                    let self_as_u8 = (self as *const Ipv6Header) as *const u8;
-                    let port_as_u8 = self_as_u8.offset(self.offset() as isize);
+            unsafe {
+                let self_as_u8 = (self as *const Ipv6Header) as *const u8;
+
+                let mut next_hdr = next_hdr as u8;
+                let mut payload_offset = 0; //track to make sure we don't go beyond v6 payload size
+
+                while next_hdr != TCP_NXT_HDR && next_hdr != UDP_NXT_HDR && self.payload_size(0) > payload_offset {
+                    let seek_to = self_as_u8.offset((self.offset() + payload_offset) as isize);
+                    next_hdr = seek_to.read();
+                    payload_offset += seek_to.offset(1).read() as usize * 8 + 8; //skips the current ext header
```

Comment:
can you drop in a quick comment about the math here?

Reply:
Added more comments.